include/libdnf/rpm/repo.hpp

```diff
@@ -118,262 +109,349 @@ struct Repo {
+    static void download_url(ConfigMain * cfg, const char * url, int fd);
 };
 
-/**
-* @class PackageTargetCB
-*
-* @brief Base class for PackageTarget callbacks
-*
-* User implements PackageTarget callbacks by inheriting this class and overriding its methods.
-*/
+/// Base class for PackageTarget callbacks
+/// User implements PackageTarget callbacks by inheriting this class and overriding its methods.
 class PackageTargetCB {
 public:
-    /** Transfer status codes */
-    enum class TransferStatus {
-        SUCCESSFUL,
-        ALREADYEXISTS,
-        ERROR
-    };
+    // Transfer status codes
+    enum class TransferStatus { SUCCESSFUL, ALREADYEXISTS, ERROR };
 
     virtual int end(TransferStatus status, const char * msg);
-    virtual int progress(double totalToDownload, double downloaded);
-    virtual int mirrorFailure(const char *msg, const char *url);
+    virtual int progress(double total_to_download, double downloaded);
+    virtual int mirror_failure(const char * msg, const char * url);
     virtual ~PackageTargetCB() = default;
 };
 
-/**
-* @class PackageTarget
-*
-* @brief Wraps librepo PackageTarget
-*/
+/// Wraps librepo PackageTarget
 struct PackageTarget {
 public:
-    /** Enum of supported checksum types.
-    * NOTE! This enum guarantee to be sorted by "hash quality"
-    */
+    // Enum of supported checksum types.
+    // NOTE! This enum guarantee to be sorted by "hash quality"
     enum class ChecksumType {
         UNKNOWN,
-        MD5,        /*    The most weakest hash */
-        SHA1,       /*  |                       */
-        SHA224,     /*  |                       */
-        SHA256,     /*  |                       */
-        SHA384,     /* \|/                      */
-        SHA512,     /*    The most secure hash  */
+        MD5,     //    The most weakest hash
+        SHA1,    //  |
+        SHA224,  //  |
+        SHA256,  //  |
+        SHA384,  // \|/
+        SHA512,  //    The most secure hash
     };
 
-    static ChecksumType checksumType(const std::string & name);
-    static void downloadPackages(std::vector<PackageTarget *> & targets, bool failFast);
-
-    PackageTarget(Repo * repo, const char * relativeUrl, const char * dest, int chksType,
-                  const char * chksum, int64_t expectedSize, const char * baseUrl, bool resume,
-                  int64_t byteRangeStart, int64_t byteRangeEnd, PackageTargetCB * callbacks);
-    PackageTarget(ConfigMain * cfg, const char * relativeUrl, const char * dest, int chksType,
-                  const char * chksum, int64_t expectedSize, const char * baseUrl, bool resume,
-                  int64_t byteRangeStart, int64_t byteRangeEnd, PackageTargetCB * callbacks,
-                  const char * httpHeaders[] = nullptr);
+    /// @replaces libdnf:repo/Repo.hpp:method:PacketTarget.checksumType(const std::string & name)
+    static ChecksumType checksum_type(const std::string & name);
```

Comment:
Do you mean that name represents a string form of checksum? Probably we can use
```
const char *solv_chksum_type2str(Id type);
Id solv_chksum_str2type(const char *str);
```

Reply:
`lr_checksum_type()` from librepo is used internaly. I must compare it to functions from libsolv.

Reply:
I looked on it and prefer rework in another PR. I will addd TODO.